bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -434,9 +441,8 @@ private void processStateChangedEvent(SomfyTahomaEvent event) {
 
     private void refreshTahomaStates() {
         logger.debug("Refreshing Tahoma states...");
-        if (ThingStatus.OFFLINE.equals(thing.getStatus())) {
-            logger.debug("Doing relogin");
-            login();
+        if (ThingStatus.OFFLINE.equals(thing.getStatus()) && !reLogin()) {
```

Comment:
Because `ThingStatus` is an enum so you can use == in comparisons:

```suggestion
        if (ThingStatus.OFFLINE == thing.getStatus() && !reLogin()) {
```

Reply:
changed